pkg/controller/atlasproject/private_endpoint.go

```diff
@@ -516,9 +530,16 @@ func terminateWithError(ctx *workflow.Context, conditionType api.ConditionType,
 var notReadyServiceResult = workflow.InProgress(workflow.ProjectPEServiceIsNotReadyInAtlas, "Private Endpoint Service is not ready")
 var notReadyInterfaceResult = workflow.InProgress(workflow.ProjectPEInterfaceIsNotReadyInAtlas, "Interface Private Endpoint is not ready")
 
-func getEndpointsNotInSpec(specPEs []akov2.PrivateEndpoint, atlasPEs []atlasPE) []atlasPE {
-	uniqueItems, _ := getUniqueDifference(atlasPEs, specPEs)
-	return uniqueItems
+func getEndpointsNotInSpec(specPEs []akov2.PrivateEndpoint, atlasPEs []atlasPE, lastAppliedPEs map[string]akov2.PrivateEndpoint) []atlasPE {
+	notInSpecItems, _ := getUniqueDifference(atlasPEs, specPEs)
+	toDelete := make([]atlasPE, 0, len(notInSpecItems))
+	for _, item := range notInSpecItems {
+		if _, ok := lastAppliedPEs[item.Identifier().(string)]; ok {
```

Comment:
this uses the deprecated `DeprecatedIdentifiable` interface. we should move this sort of comparisons in the translation layer and not rely on deprecated functionality.

Reply:
Again, this sub-resource doesn't use translation layer. Unless we migrate everything to translation layer, it's not possible to get rid of `DeprecatedIdentifiable`

Reply:
indeed, resolving, thank you!